sdk_extensions/aws_v1_support/src/main/java/io/opentelemetry/sdk/extensions/trace/aws/resource/JdkHttpClient.java

```diff
@@ -0,0 +1,163 @@
+
+  private static final int TIMEOUT_MILLIS = 2000;
+
+  String fetchString(
+      String httpMethod,
+      String urlStr,
+      Map<String, String> requestPropertyMap,
+      String keyStorePath) {
+    final HttpURLConnection connection;
+
+    try {
+      if (urlStr.startsWith("https")) {
+        connection = (HttpsURLConnection) new URL(urlStr).openConnection();
+        KeyStore keyStore = getK8sKeystore(keyStorePath);
+        if (keyStore != null) {
+          ((HttpsURLConnection) connection).setSSLSocketFactory(buildSslSocketFactory(keyStore));
+        }
+      } else {
+        connection = (HttpURLConnection) new URL(urlStr).openConnection();
+      }
+
+      connection.setRequestMethod(httpMethod);
+      connection.setConnectTimeout(TIMEOUT_MILLIS);
+      connection.setReadTimeout(TIMEOUT_MILLIS);
+
+      for (Map.Entry<String, String> requestProperty : requestPropertyMap.entrySet()) {
+        connection.setRequestProperty(requestProperty.getKey(), requestProperty.getValue());
+      }
+
+      int responseCode = connection.getResponseCode();
+      if (responseCode != 200) {
+        logger.log(
+            Level.FINE,
+            "Error reponse from "
+                + urlStr
+                + " code ("
+                + responseCode
+                + ") text "
+                + readResponseString(connection));
+        return "";
+      }
+
+      return readResponseString(connection).trim();
+
+    } catch (IOException e) {
+      logger.log(Level.WARNING, "JdkHttpClient fetch string failed: " + e);
+    }
+
+    return "";
+  }
+
+  private static String readResponseString(HttpURLConnection connection) {
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+    try (InputStream is = connection.getInputStream()) {
+      ByteStreams.copy(is, os);
+    } catch (IOException e) {
+      // Only best effort read if we can.
+    }
+    try (InputStream is = connection.getErrorStream()) {
+      if (is != null) {
+        ByteStreams.copy(is, os);
+      }
+    } catch (IOException e) {
+      // Only best effort read if we can.
+    }
+    try {
+      return os.toString(StandardCharsets.UTF_8.name());
+    } catch (UnsupportedEncodingException e) {
+      logger.log(Level.WARNING, "UTF-8 not supported can't happen. " + e);
+    }
+    return "";
+  }
+
+  private static SSLSocketFactory buildSslSocketFactory(KeyStore keyStore) {
+    try {
+      String tmfAlgorithm = TrustManagerFactory.getDefaultAlgorithm();
+      TrustManagerFactory tmf = TrustManagerFactory.getInstance(tmfAlgorithm);
+      tmf.init(keyStore);
+
+      SSLContext context = SSLContext.getInstance("TLS");
+      context.init(null, tmf.getTrustManagers(), null);
+      return context.getSocketFactory();
+
+    } catch (Exception e) {
+      logger.log(Level.WARNING, "Build SslSocketFactory for K8s restful client exception: " + e);
+    }
+    return null;
+  }
+
+  String getK8sCredHeader(String tokenFilePath) {
```

Comment:
Move this into `EksResource` it's k8s specific

Reply:
move method to `EksResource`